src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -40,13 +42,19 @@
 
     private final PloidyModel ploidyModel;
 
+    private final int MAX_GENOTYPE_COUNT_TO_ENUMERATE = configuration.genotypeArgs.MAX_GENOTYPE_COUNT;
```

Comment:
This is a bit strange since it's relying on the constructor ordering in a bit of a strange way.  I think you'd be better off initializing this in the constructor where it's more obviously clear that it won't result in a null ptr dereference.